next-cloudinary/src/lib/cloudinary.js

```diff
@@ -104,6 +104,28 @@ export function getPublicId(src) {
 
   return src;
 }
+export function getTransformations(src, useUrlTransformations) {
+  if ( typeof src !== 'string' ) {
+    throw new Error(`Invalid src of type ${typeof src}`);
+  }
+
+  if (src.includes('res.cloudinary.com') && useUrlTransformations) {
+    const regex = new RegExp(
+      "(https?)://(res.cloudinary.com)/([^/]+)/(image|video|raw)/(upload|authenticated)/(.*)/(v[0-9]+)/(.+)(?:.[a-z]{3})?",
```

Comment:
is this the same regex as one of the existing ones? i wonder if it makes sense eto cache the regex instances, as iirc these can be expensive and scaling up could cause issues

Reply:
The way I know how to cache it is to use the package "regex-cache". I don't know if you're up for using a dependency for this feature. The link to the package is [this](https://www.npmjs.com/package/regex-cache) and it is pretty trusted.